dali/pipeline/operator/builtin/external_source.cc

```diff
@@ -97,6 +97,11 @@ The operator will validate that the fetched data is of the provided type.
 If the argument is omitted or ``DALIDataType.NO_TYPE`` is passed, the operator will infer
 the type based on the provided data.
 
-This argument will be required starting from DALI 2.0.)code", DALI_NO_TYPE);
+This argument will be required starting from DALI 2.0.)code", DALI_NO_TYPE)
```

Comment:
A high level remark:
How about making `layout` an argument, too? This way you could cross-validate ndim/layout here - and even infer ndim from the layout string.

Reply:
I agree that `layout` should be also a C++-level argument - even just because we tend to avoid Python-only features. Although, I would make this another task maybe. I want to move on with the rest of things needed for the Triton backend autofill. After doing that I can revisit the ES and make layout a C++ arg.